pkg/scraper/client/resource/client.go

```diff
@@ -104,33 +85,64 @@ func (kc *kubeletClient) GetMetrics(ctx context.Context, node *corev1.Node) (*st
 	}
 	addr, err := kc.addrResolver.NodeAddress(node)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to extract connection information for node %q: %v", node.Name, err)
 	}
 	url := url.URL{
-		Scheme:   kc.scheme,
-		Host:     net.JoinHostPort(addr, strconv.Itoa(port)),
-		Path:     "/stats/summary",
-		RawQuery: "only_cpu_and_memory=true",
+		Scheme: kc.scheme,
+		Host:   net.JoinHostPort(addr, strconv.Itoa(port)),
+		Path:   "/metrics/resource",
 	}
 
 	req, err := http.NewRequest("GET", url.String(), nil)
 	if err != nil {
 		return nil, err
 	}
-	summary := &Summary{}
-	client := kc.client
-	if client == nil {
-		client = http.DefaultClient
+	//1. Send request, fetch data from kubelet/metrics/resource endpoint
```

Comment:
Instead of leaving comments that can get outdated, let's improve names of functions.
Maybe instead of makeRequest function, we could rename it to sendRequest instead of leaving comment "send request"